### PR TITLE
fix: static file support route_prefix

### PR DESCRIPTION
**Pull request by solarknight**
Use purely random route for static files if --random-route enabled, otherwise add route_prefix before generated route

### CODE REVIEW

**Comment by svenstaro** [comment]
Thanks! Any chance you could also make sure this is tested by one of the tests? Otherwise we might get regressions again.

**Comment by solarknight** [comment]
> Thanks! Any chance you could also make sure this is tested by one of the tests? Otherwise we might get regressions again.
Oh sorry, I'll check the tests ：）

**Comment by svenstaro** [comment]
Also check out the CI which failed due to some formatting problem. :)

**Comment by svenstaro** [comment]
@solarknight Hey, you still looking into this.

**Comment by solarknight** [comment]
> 
Hey, finally I complete the test for static file route check (really a newbie to rust). 
Unfortunately I use push -f by mistake. Do you mind if I make a new PR, or any other suggestions?

**Comment by svenstaro** [comment]
Could you rebase this so the history becomes tidier?

**Review by svenstaro — CHANGES_REQUESTED** [review]

File: src/config.rs [hunk 1]
```diff
@@ -151,8 +151,17 @@ impl MiniserveConfig {
...
+            (
+                format!("/{}", nanoid::nanoid!(10, &ROUTE_ALPHABET)),
+                format!("/{}", nanoid::nanoid!(10, &ROUTE_ALPHABET)),
+            )
+        } else {
+            (
+                format!("{}/{}", route_prefix, nanoid::nanoid!(10, &ROUTE_ALPHABET)),
+                format!("{}/{}", route_prefix, nanoid::nanoid!(10, &ROUTE_ALPHABET)),
+            )
+        };
```
Comment (svenstaro) [inline src/config.rs:164]: Please expand the comment above and describe these two cases and why it's necessary do handle it like this.

File: tests/config.rs [hunk 2]
```diff
@@ -0,0 +1,28 @@
+mod fixtures;
+
+use fixtures::{server_no_stderr, Error, TestServer};
+use regex::Regex;
+use rstest::rstest;
+use select::{document::Document, predicate::Attr};
+
+#[rstest]
```
[the remark this answers is not in the document]
Comment (solarknight) [reply tests/config.rs]: Done, rename the test to keep consistent with existing tests.

File: src/config.rs (hunk 1 above)
Comment (solarknight) [reply src/config.rs:167]: Done. please check the comment.

**Comment by solarknight** [comment]
@svenstaro All above suggestions should be applied, please review the changes.

**Review by svenstaro — APPROVED** [review]

**Comment by svenstaro** [comment]
Very cool, thanks!

**Pull request merged**